interface/wx/memoryallocator.h

```diff
@@ -0,0 +1,98 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        memoryallocator.h
+// Purpose:     interface of wxMemoryAllocator
+// Author:      wxWidgets team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+/**
+    @class wxMemoryAllocator
+
+    This is a little helper class that hides the difference between
+    C++ operators new/delete and C way of memory management using
+    malloc()/free(). It is currently being used in wxImage to do
+    memory management. Right now it has 3 direct descendants - 1
+    for C-way, 1 for C++-way and 1 that will not throw on allocation.
+
+    @since 3.1.1
+*/
+class wxMemoryAllocator
+{
+public:
+    /**
+        Destructor.
+    */
+    virtual ~wxMemoryAllocator();
+    
+    /**
+        Default memory allocation function. This function may
+        either return NULL or throw std::bad_alloc on error.
+    */
+    void *Alloc(size_t size) = 0;
```

Comment:
I believe all virtual methods should be always labeled as such. Applies to all four classes.

Reply:
@PBfordev ,
I think for internal virtual functions this is not correct.
But I will fix it and let the approving party decide.